docker/Dockerfile

```diff
@@ -93,46 +89,76 @@ RUN python /usr/local/generate_toolchain.py --prefix /usr/local
 RUN install-from-source https://www.zlib.net/zlib-1.2.11.tar.gz
 
 # LLVM and Clang
-RUN BUILD_FILE_DIR='llvm' install-cmake-from-source \
-    https://github.com/llvm/llvm-project/archive/llvmorg-9.0.1.tar.gz \
-    -DLLVM_ENABLE_PROJECTS="clang"
-
-# Libpci access and libdrm for communicating with graphics hardware
-RUN install-from-source https://xorg.freedesktop.org/releases/individual/lib/libpciaccess-0.16.tar.bz2
-RUN install-meson-from-source https://dri.freedesktop.org/libdrm/libdrm-2.4.100.tar.bz2 \
-    -Dudev=false \
-    -Dvalgrind=false
-
-# Intel graphics compiler
-RUN install-cmake-from-source https://github.com/KhronosGroup/SPIRV-LLVM-Translator/archive/v9.0.0-1.tar.gz
+RUN BUILD_FILE_DIR='llvm' install-cmake-from-source https://github.com/llvm/llvm-project/archive/llvmorg-11.0.1.tar.gz \
+    -DLLVM_ENABLE_PROJECTS="clang" \
+    -DLLVM_APPEND_VC_REV=ON \
+    -DLLVM_ENABLE_RTTI=ON \
+    -DLLVM_ENABLE_FFI=ON \
+    -DFFI_INCLUDE_DIR="$(pkg-config --variable=includedir libffi)" \
+    -DFFI_LIBRARY_DIR="$(pkg-config --variable=libdir libffi)" \
+    -DLLVM_ENABLE_SPHINX=OFF \
+    -DLLVM_BUILD_LLVM_DYLIB=ON \
+    -DLLVM_LINK_LLVM_DYLIB=ON \
+    -DLLVM_BINUTILS_INCDIR=/usr/include
+
+# SPIRV LLVM Translator
+RUN install-cmake-from-source https://github.com/KhronosGroup/SPIRV-LLVM-Translator/archive/v11.0.0.tar.gz \
+    -DBUILD_SHARED_LIBS=ON \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DCMAKE_SKIP_RPATH=ON \
+    -DLLVM_INCLUDE_TESTS=ON \
+    -DLLVM_EXTERNAL_LIT=/usr/bin/lit
```

Comment:
`/usr/local/bin/lit`? This comes from LLVM right?

Reply:
That's one of the flags from the arch PKGBUILD which I grabbed. Now that I think I have a fix for the intel compute runtime building issues I found before, should I remove those flags which I added which weren't necessary? Other examples include all of these 4 flags in this snippet, and things like `-DLLVM_BUILD_LLVM_DYLIB=ON`

If not, in this specific case, I am not sure. It would seem to me that we could get by without that flag or the one above it which includes tests.

Reply:
I would remove it, and probably the test one too. Building llvm is painful enough without building the tests

Reply:
If you think they are unnecessary then remove them.

Including tests is something we almost always turn off to speed up the builds. 

We need to be careful with our builds, specifically the NUC builds. You need to assume that the NUC build is not capable of running on the host machine. So you should assume that running `lit` from a LLVM that was built for the NUC is going to fail. In this case, we should be installing LLVM/Clang from pacman and using `-DLLVM_EXTERNAL_LIT=/usr/bin/lit` to ensure that it will work

Reply:
I have removed those two and all the others I added which weren't required or didn't remove stuff we didn't use

Reply:
@Bidski could you clarify that a little please? when exactly should we install llvm/clang with pacman?

Reply:
There should be a system installation of LLVM/Clang, so that when you build SPIRV-LLVM-Translator you can direct it to link against the LLVM/Clang that we built, but use `lit` (and other binaries) while it is building.

This needs to be the case for any library that needs to run a binary that it compiles as part of its build process (this is the massive headache that is cross-compilation). For instance, in `opencl-clang.sh` we first build `linux_resource_linker` without any special compiler flags so that `opencl-clang` can safely run that binary while it is building.

Reply:
Jeeez  that sounds like an absolute headache! I'll note that for a future lil' project for the build team :wink: 
Are there other packages which could use similar treatment?

Reply:
This is mostly a trial and error thing, unless you study the build process of every library we build.

We have covered most of the libraries we install already, but we may have missed some. At this point, I would just wait until some poor unlucky soul gets a bizarre build issue :P

Reply:
hahahaha roger that